router/function_test.go

```diff
@@ -133,9 +133,78 @@ var _ = Describe("Function", func() {
 
 	})
 
+	Context("with a GoFormation AWS::Serverless::Function and prefix matching", func() {
+
+		r := router.NewServerlessRouter(true)
+
+		function := &cloudformation.AWSServerlessFunction{
+			Runtime: "nodejs6.10",
+			Events: map[string]cloudformation.AWSServerlessFunction_EventSource{
+				"AnyRequests": cloudformation.AWSServerlessFunction_EventSource{
+					Type: "Api",
+					Properties: &cloudformation.AWSServerlessFunction_S3EventOrSNSEventOrKinesisEventOrDynamoDBEventOrApiEventOrScheduleEventOrCloudWatchEventEventOrIoTRuleEventOrAlexaSkillEvent{
+						ApiEvent: &cloudformation.AWSServerlessFunction_ApiEvent{
+							Path:   "/any",
```

Comment:
Actually, prefix routing should be automatically enabled when route is `/foo/bar/{path}+`. See: https://aws.amazon.com/blogs/aws/api-gateway-update-new-features-simplify-api-development/. 

It should be fairly simple change to make: In `router.go`, just check for path containing catch-alls and enable prefix routing on the prefix ie. `/foo/bar/{path}+` would enable prefix routing on `/foo/bar`

Reply:
Updated pull request to include this functionality. Basically if we detect that a path ends with a catch all variables (e.g. `{foo+}` we instruct the mount to use prefix matching.

I added this functionality to the ServerlessRouterMount to better decouple the implementation (i.e. router) from the metadata (i.e. mount).

Let me know what you think.